### ISSUE TITLE
Java Plugin template doesn't compile

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#50480]
## Steps to reproduces:
1) `flutter create -t plugin -i objc -a java JavaTest`
1) open "JavaTest/android" with Android Studio
1) Notice the error `Minimum supported Gradle version is 5.4.1.  Current version is 4.10.2`
1) Press the 'fix and re-import` button
1) Build the project, notice it doesn't build reporting errors

## Errors:
It looks like everything stems from not being able to find `androidx.annotation.NonNull`
```
com/example/JavaTest/JavaTestPlugin.java |  
-- | --
error: cannot find symbol class NonNull |  
error: package io.flutter.embedding.engine.plugins does not exist |  
error: package io.flutter.plugin.common does not exist |  
error: package io.flutter.plugin.common does not exist |  
error: package io.flutter.plugin.common.MethodChannel does not exist |  
error: package io.flutter.plugin.common.MethodChannel does not exist |  
error: package io.flutter.plugin.common.PluginRegistry does not exist |  
```

## Notes:
- I tried to run the refactor tool to migrate to androidx and it performed a no-op

## Doctor:
```
[✓] Flutter (Channel master, v1.14.6-pre.237, on Mac OS X 10.14.6 18G2022, locale en-US)
    • Flutter version 1.14.6-pre.237 at /Users/aaclarke/dev/flutter
    • Framework revision 6dc3bfaa98 (3 days ago), 2020-02-07 18:28:02 -0800
    • Engine revision 6158f03ef5
    • Dart version 2.8.0 (build 2.8.0-dev.8.0 514a8d4c84)
```

cc @blasten

**Comment by gaaclarke** [comment]
Note: adding `implementation 'androidx.annotation:annotation:1.1.0'` to the build.gradle got me past this error, but the next import for embedding wasn't found after that.

**Comment by gaaclarke** [comment]
Note: I was compiling from `JavaTest/android/`, compiling from `JavaTest/example/android` works.

**Comment by dnfield** [comment]
This should be fixed by #50388 

That said, we might be breaking other people who don't work with the current version - @blasten is that true?

**Comment by blasten** [comment]
@dnfield I don't think so. #50388 is fixing a bug in the template, but the impact is reduced to:
1. Running `flutter build aar` on a plugin project.
2. Open the plugin with Android Studio and building the plugin individually. (This issue).

In most cases, people test plugins using the example app.
